tensorflow/stream_executor/dnn.h

```diff
@@ -667,8 +667,31 @@ class PoolingDescriptor {
   std::vector<int64> strides_;
 };
 
-typedef int64 AlgorithmType;
-constexpr AlgorithmType kDefaultAlgorithm = -1;
+class AlgorithmType {
+  public:
+  typedef int64 Index;
+  AlgorithmType()
+      : algo_(kDefaultAlgorithm), tensor_ops_enabled_(false) {}
+  AlgorithmType(Index a, bool use_tensor_ops)
+      : algo_(a), tensor_ops_enabled_(use_tensor_ops) {}
+  bool is_default() const { return algo_ == kDefaultAlgorithm; }
+  bool operator==(const AlgorithmType& other) const {
+    return this->algo_ == other.algo_ &&
+           this->tensor_ops_enabled_ == other.tensor_ops_enabled_;
+  }
+  bool tensor_ops_enabled() const {
+    return tensor_ops_enabled_;
+  }
+  Index get_algo_id() const {
```

Comment:
Suggest simply algo_id() -- that would match the other functions.

Reply:
Done